python/ray/import_thread.py

```diff
@@ -98,6 +106,51 @@ def _process_key(self, key):
             if key.startswith(b"FunctionsToRun"):
                 with profiling.profile("fetch_and_run_function"):
                     self.fetch_and_execute_function_to_run(key)
+
+            # If the same remote function or actor definition appears to be
+            # exported many times, then print a warning. We only issue this
+            # warning from the driver so that it is only triggered once instead
+            # of many times. TODO(rkn): We may want to push this to the driver
+            # through Redis so that it can be displayed in the dashboard more
+            # easily.
+            else:
+                error_message = (
+                    "The %s '%s' has been exported %s times. It's possible "
+                    "that this warning is accidental, but this may indicate "
+                    "that the same remote function is being defined "
+                    "repeatedly from within many tasks and exported to all of "
+                    "the workers. This can be a performance issue and can be "
+                    "resolved by defining the remote function on the driver "
+                    "instead. See "
+                    "https://github.com/ray-project/ray/issues/6240 for more "
+                    "discussion.")
+
+                if key.startswith(b"RemoteFunction"):
+                    collision_identifier, function_name = (
+                        self.redis_client.hmget(
+                            key, ["collision_identifier", "name"]))
```

Comment:
```suggestion
                            key, ["collision_identifier", "function_name"]))
```
This would be more clear and matches `class_name` used for actors. If this requires making many changes, don't worry about it.